docs/topics/proc_starting-your-application-on-openshift-in-debugging-mode.adoc

```diff
@@ -0,0 +1,64 @@
+
+[#starting-your-application-on-openshift-in-debugging-mode_{context}]
+= Starting Your Application on OpenShift in Debugging Mode
+
+To debug your :runtime:-based application on OpenShift remotely, you must set the `JAVA_DEBUG` environment variable inside the container to `true` and configure port forwarding so that you can connect to your application from a remote debugger.
+
+.Prerequisites
+
+* Your application running on OpenShift.
+* The `oc` binary installed on your machine.
+
+.Procedure
+
+. Using the `oc` command, list the available deployment configurations:
++
+[source,bash]
+----
+$ oc get dc
+----
+
+. Set the `JAVA_DEBUG` environment variable in the deployment configuration of your application to `true`, which configures the JVM to open the port number 5005 for debugging. For example:
++
+[source,bash]
+----
+$ oc set env dc/my-application JAVA_DEBUG=true
+----
+
+. Redeploy the application if it is not set to redeploy automatically on environment change. For example:
++
+[source,bash]
+----
+$ oc rollout latest dc/my-application
+----
+
+. Configure port forwarding from the application pod:
+.. List the currently running pods and find one containing your application:
++
+[source,bash,options="nowrap"]
+----
+$ oc get pod
+NAME                            READY     STATUS      RESTARTS   AGE
+my-application-3-1xrsp          0/1       Running     0          6s
+...
+----
+
+.. Configure port forwarding from the application pod:
++
+--
+[source,bash,options="nowrap"]
+----
+$ oc port-forward my-application-3-1xrsp $LOCAL_PORT_NUMBER:5005
+----
+
+Here, `$LOCAL_PORT_NUMBER` is an unused port number of your choice.
+Remember this number for the remote debugger configuration.
+--
+
+. When you are done debugging, disable the `JAVA_DEBUG` environment variable in your application pod. For example:
```

Comment:
The syntax `oc set env dc/my-app JAVA_DEBUG-` doesn't "disable" the environment variable, it _removes_ it. It would be possible to _disable_ remote debugging by setting the `JAVA_DEBUG` env var to `false`, but that's not what we're doing here.